### PR TITLE
[feat]: Add support for gpt-oss

### PR DESCRIPTION
**Pull request by yeshsurya**
## Summary
This work adds support for gpt oss models

## Testing Done: 

Inference Benchmark Execution 
Total inference runs: 18 executions
12 successful benchmark measurements
6 pre-run warmup iterations
Configurations tested: 3 scenarios with 2 runs each
All scenarios passed with consistent, measurable improvements

- Hardware Type: RTX A6000
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/model/gpt_oss.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    shift_labels = kwargs.pop("shift_labels", None)
+    logits = None
+    loss = None
+
+    if skip_logits is None:
+        skip_logits = self.training and (labels is not None or shift_labels is not None)
+
+    if skip_logits:
+        loss = LigerForCausalLMLoss(
```
Comment (kashif) [inline src/liger_kernel/transformers/model/gpt_oss.py]: can you kindly check the other model examples since the LigerForCausalLMLoss can return a tuple if return_token_accuracy=True in the kwargs so just do as in the other model examples

File: src/liger_kernel/transformers/model/gpt_oss.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    >>> tokenizer = AutoTokenizer.from_pretrained("openai/gpt-oss-20b")
+
+    >>> prompt = "Hey, are you conscious? Can you talk to me?"
+    >>> inputs = tokenizer(prompt, return_tensors="pt")
+
+    >>> # Generate
+    >>> generate_ids = model.generate(inputs.input_ids, max_length=30)
+    >>> tokenizer.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]
+    "Hey, are you conscious? Can you talk to me?\nI'm not conscious, but I can talk to you."
+    ```"""
```
Comment (kashif) [inline src/liger_kernel/transformers/model/gpt_oss.py:131]: this does not seem to match the output of this function

Comment (yeshsurya) [reply src/liger_kernel/transformers/model/gpt_oss.py:131]: Updated doc string; But in other model files they seem to have an example similar to previous one I had. ( refering : gemma.py )

**Comment by yeshsurya** [comment]
@Tcc0403, help review

**Comment by yeshsurya** [comment]
@shimizust , help review

**Comment by yeshsurya** [comment]
> Thank you, lgtm!

please help to merge this, looks like gpus tests are to be skipped ? since no cloud secrets are configured for them ?

**Review by Tcc0403 — APPROVED** [review]
